crypto/legacy_crypto/src/x25519.rs

```diff
@@ -89,6 +91,14 @@ impl Clone for X25519PrivateKey {
     }
 }
 
+impl Deref for X25519PrivateKey {
```

Comment:
IMO we should better get rid of `impl Deref` everywhere in crypto crate, nobody should have access on the backing key structure.

Reply:
I agree - with this PR I just wanted to get rid of the dependency without touching functionality.

Reply:
Right, approving as scope wise it makes sense.

Reply:
@kchalkias  `Deref` on `X25519StaticFoo` is *very* necessary and intended, for all values of `Foo`, in a way that's really specific to X25519 Ephemeral / Static types. It should not be removed.